packages/ERTP/src/paymentLedger.js

```diff
@@ -0,0 +1,267 @@
+    return E.when(paymentP, payment => {
+      assertKnownPayment(payment);
+      const paymentBalance = paymentLedger.get(payment);
+      assertAmountEqual(paymentBalance, optAmount);
+      // Commit point.
+      paymentLedger.delete(payment);
+      return paymentBalance;
+    });
+  };
+
+  const claim = (paymentP, optAmount = undefined) => {
+    return E.when(paymentP, srcPayment => {
+      assertKnownPayment(srcPayment);
+      const srcPaymentBalance = paymentLedger.get(srcPayment);
+      assertAmountEqual(srcPaymentBalance, optAmount);
+      // Commit point.
+      const [payment] = reallocate([srcPayment], [srcPaymentBalance]);
+      return payment;
+    });
+  };
+
+  // Payments in `fromPaymentsPArray` must be distinct. Alias
+  // checking is delegated to the `reallocate` function.
+  const combine = (fromPaymentsPArray, optTotalAmount = undefined) => {
+    return Promise.all(fromPaymentsPArray).then(fromPaymentsArray => {
+      fromPaymentsArray.every(assertKnownPayment);
+      const totalPaymentsBalance = fromPaymentsArray
+        .map(paymentLedger.get)
+        .reduce(add, emptyAmount);
+      assertAmountEqual(totalPaymentsBalance, optTotalAmount);
+      // Commit point.
+      const [payment] = reallocate(fromPaymentsArray, [totalPaymentsBalance]);
+      return payment;
+    });
+  };
+
+  // payment to two payments, A and B
+  const split = (paymentP, paymentAmountA) => {
+    return E.when(paymentP, srcPayment => {
+      paymentAmountA = coerce(paymentAmountA);
+      assertKnownPayment(srcPayment);
+      const srcPaymentBalance = paymentLedger.get(srcPayment);
+      const paymentAmountB = subtract(srcPaymentBalance, paymentAmountA);
+      // Commit point
+      const newPayments = reallocate(
+        [srcPayment],
+        [paymentAmountA, paymentAmountB],
+      );
+      return newPayments;
+    });
+  };
+
+  const splitMany = (paymentP, amounts) => {
+    return E.when(paymentP, srcPayment => {
+      assertKnownPayment(srcPayment);
+      amounts = amounts.map(coerce);
+      // Commit point
+      const newPayments = reallocate([srcPayment], amounts);
+      return newPayments;
+    });
+  };
+
+  const mintPayment = newAmount => {
+    newAmount = coerce(newAmount);
+    const payment = makePayment();
+    paymentLedger.init(payment, newAmount);
+    return payment;
+  };
+
+  const deposit = (
+    currentBalance,
+    commit,
+    srcPayment,
+    optAmount = undefined,
+  ) => {
+    if (isPromise(srcPayment)) {
+      throw TypeError(
+        `deposit does not accept promises as first argument. Instead of passing the promise (deposit(paymentPromise)), consider unwrapping the promise first: paymentPromise.then(actualPayment => deposit(actualPayment))`,
+      );
+    }
+    assertKnownPayment(srcPayment);
+    const srcPaymentBalance = paymentLedger.get(srcPayment);
+    // Note: this does not guarantee that optAmount itself is a valid stable amount
```

Comment:
```suggestion
    // Note: this does not guarantee that optAmount itself is a valid stable 
    // amount
```

Please check my understanding: This assert does guarantee that `optAmount` is either `undefined` or a valid stable amount consistent with the brand, yes?

Reply:
I believe I added that comment at your direction, to make sure we didn't use the user-provided `optAmount` later in the uncoerced form:

https://github.com/Agoric/agoric-sdk/pull/523/files#r382361674

The link wasn't reliably getting to the comment for me, so here's a screenshot too:

<img width="970" alt="Screen Shot 2021-05-20 at 6 55 52 PM" src="https://user-images.githubusercontent.com/2441069/119070597-2b285680-b99d-11eb-90f3-bd4f5481402d.png">

What do you think? Should we change the comment? Remove it?

Reply:
Keep the comment. I underestimated it now and will have to think more about my earlier worries. Thanks for the reminder!

I may suggest a rewording but don't worry about that now. Certainly, this PR need not wait on such a rewording.

Reply:
No change suggested. Resolving.